bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
Above I wrote

* Range1d
   * start - Either(Float, Datetime, Int)
   * start_bound - Either(Float, Datetime, Int, Auto, None, default=??)
   * end - Either(Float, Datetime, Int)
   * end_bound - Either(Float, Datetime, Int, Auto, None, default=??)

But I'd like to ammend that to 

* Range1d
   * start - Either(Float, Datetime, Int)
   * start_bound - Either(Float, Datetime, Int, Auto, None, default=Auto)
   * end - Either(Float, Datetime, Int)
   * end_bound - Either(Float, Datetime, Int, Auto, None, default=Auto)

Having `default=Auto` will lead to a change in behavior from what users are currently used to with Bokeh. However, I think this is ok because:

1) We're still going through some API churn so it's an ok time to make a change
2) If you really do want None I think it's easier to spell than spelling Auto which I think lots of people will want. See below.

If default is Auto and you want None:

```
x_range = Range1d(start=1, end=2, start_bound=None, end_bound=None)
```

If default is None and you want Auto:
```python
from bokeh.properties import Auto
x_range = Range1d(start=1, end=2, start_bound=Auto, end_bound=Auto)
```